<launch>

  <arg name="rviz" default="true" />
  <arg name="scanPeriod" default="0.1" />
  <arg name="lidarName" default="PandarQT" />
  <arg name="pointCloudName" default="PandarQT_Data" />

  <node pkg="loam_velodyne" type="multiScanRegistration" name="multiScanRegistration" output="screen">
    <param name="lidar" value="$(arg lidarName)" /> <!-- options: VLP-16  HDL-32  HDL-64E PandarQT-->
    <param name="scanPeriod" value="$(arg scanPeriod)" />
    <param name="PointCloudTopicName" value="$(arg pointCloudName)" />
  </node>

  <node pkg="loam_velodyne" type="laserOdometry" name="laserOdometry" output="screen" respawn="true">
    <param name="scanPeriod" value="$(arg scanPeriod)" />
  </node>

  <node pkg="loam_velodyne" type="laserMapping" name="laserMapping" output="screen">
    <param name="scanPeriod" value="$(arg scanPeriod)" />
  </node>

  <node pkg="loam_velodyne" type="transformMaintenance" name="transformMaintenance" output="screen">
  </node>

  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find loam_velodyne)/rviz_cfg/loam_velodyne.rviz" />
  </group>

</launch>
